pages/profile/main.js

```diff
@@ -108,11 +108,10 @@ var profilePageInit = async function () {
    */
   async function fetchProfileLog () {
 
-    // Fetch the actual profile log
-    const request = await fetch(`/profiles/${profileSteamID}/profile.log`);
+    // Fetch the decoded profile log
```

Comment:
The "actual" profile log was being fetched on purpose here. The `.log` file is tiny, the JSON resulting from decoding it can often be far from that. To save bandwidth (and reallocate a tiny bit of compute to the client), this should be reverted. Having an API endpoint that serves the binary file is perfectly fine.

Reply:
I was thinking more about the deduplication here (and also coincidentally forgot to remove the function, lol).
Not having duplicated the decode function would allow you to change the log with ease.
If you prefer the client fetching the log trading off duplicated code for efficiency, just confirm that and I'll revert/adjust this change

Reply:
I'm not super happy about duplication either, but IMO it's better than sending the whole JSON pointlessly.

Here's a silly thought - what if this log decompilation (along with other stuff maybe) was moved into a client-accessible file? Could even put it in tools.js - as long as you put something like `if ("module" in this) module.exports = decodeLog;` at the end, it could be used for both client and server. Though I admit that's both hacky and also going against the goal of this PR to separate these environments.

If you see a nice way of deduplicating, by all means go ahead. But sending the JSON every time does not get my support. The endpoint can remain for all I care, let's just not use it on the frontend.